src/main/java/org/cryptomator/windows/uiappearance/WinUiAppearanceProvider.java

```diff
@@ -0,0 +1,58 @@
+package org.cryptomator.windows.uiappearance;
+
+import org.cryptomator.integrations.uiappearance.Theme;
+import org.cryptomator.integrations.uiappearance.UiAppearanceException;
+import org.cryptomator.integrations.uiappearance.UiAppearanceListener;
+import org.cryptomator.integrations.uiappearance.UiAppearanceProvider;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class WinUiAppearanceProvider implements UiAppearanceProvider, UiAppearanceListener {
```

Comment:
I'm not sure if the WinUIAppearanceProvider should implement UIAppearanceListener, because it is possible to create an inite loop:

1. Create new WinUiAppearanceProvider: `var infiniteProvider = new WinUiAppearanceProvider()`
2. Register the Provider to himself: `infiniteProvider.addListener(infiniteProvider)`
3. Change the observed registry value (e.g. by using the settings panel)
4. Infinite loop: Native code calls `systemAppearanceChanged(...)`, which calls every listener. Since `infiniteProvider` listens to himself, it calls  `infiniteProvider::systemAppearanceChanged`, which calls every listener. Since the `infiniteProvider` listens to himself, it calls ...

Reply:
Good point. Will change it. There is no need for using an interface here, anyway.

Reply:
fixed in f9030b6